src/query/src/optimizer.rs

```diff
@@ -34,19 +34,15 @@ use datatypes::arrow::datatypes::DataType;
 pub struct TypeConversionRule;
 
 impl OptimizerRule for TypeConversionRule {
-    fn optimize(
-        &self,
-        plan: &LogicalPlan,
-        optimizer_config: &mut OptimizerConfig,
-    ) -> Result<LogicalPlan> {
+    fn optimize(&self, plan: &LogicalPlan, _config: &mut OptimizerConfig) -> Result<LogicalPlan> {
```

Comment:
Since `_config` is used in the following codes, we should rename it to `config`.

Reply:
clippy complained with https://rust-lang.github.io/rust-clippy/master/index.html#only_used_in_recursion

Reply:
Thanks, I get it. Looks like we could move the optimization logic to a function which only takes the plan as input, so we don't need to passing the config to it. Let's defer it and merge this PR first.